gobblin-core/src/main/java/gobblin/source/extractor/watermark/TimestampWatermark.java

```diff
@@ -46,6 +47,7 @@
   public TimestampWatermark(String watermarkColumn, String watermarkFormat) {
     this.watermarkColumn = watermarkColumn;
     this.watermarkFormat = watermarkFormat;
+    _inputFormatParser = new SimpleDateFormat(INPUTFORMAT);
```

Comment:
This approach doesn't solve the multithreading issue. What about two threads concurrently executing `getIntervals`? The right fix is to define it as a local variable or use `synchronize` primitive. I think local variable is a better way.